File: iosApp/iosApp/p2p/OfferMessageNotifiable.swift
//
//  OfferMessageNotifiable.swift
//  iosApp
//
//  Created by jimmyt on 6/10/22.
//  Copyright © 2022 orgName. All rights reserved.
//

/**
 A protocol that a structure or class must adopt in order to receive offer-related messages from `P2PService`.
 */
protocol OfferMessageNotifiable {
    /**
     The function called by `P2PService` in order to notify the structure or class adopting this protocol of a new `PublicKeyAnnouncement`.
     
     - Parameter message: the new `PublicKeyAnnouncement` of which the structure or class adopting this protocol is being notified, and should handle in the implementation of this function.
     */
    func handlePublicKeyAnnouncement(_ message: PublicKeyAnnouncement)
}

File: androidApp/src/main/java/com/commuto/interfacemobile/android/p2p/OfferMessageNotifiable.kt
package com.commuto.interfacemobile.android.p2p

import com.commuto.interfacemobile.android.p2p.messages.PublicKeyAnnouncement

/**
 * An interface that a class must implement in order to be notified of offer-related messages by
 * [P2PService].
 */
interface OfferMessageNotifiable {
    /**
     * The method called by [P2PService] in order to notify the class implementing this interface of
     * a new [PublicKeyAnnouncement].
     *
     * @param message The new [PublicKeyAnnouncement] of which the class implementing this interface
     * is being notified and should handle in the implementation of this method.
     */
    fun handlePublicKeyAnnouncement(message: PublicKeyAnnouncement)
}